src/worker.cc

```diff
@@ -62,12 +65,13 @@ Worker::~Worker() {
 }
 
 void Worker::Run() {
-  // setup gpu device
   auto context = Singleton<Context>::Instance();
+  context->SetupDevice(std::this_thread::get_id(), device_id_);
```

Comment:
It is preferred to use thread pool to setup Context (thread_id, device_id) than passing device_id as an argument of creating Worker instance.